x-pack/platform/packages/shared/kbn-elastic-assistant/impl/alerts/attack_discovery/index.tsx

```diff
@@ -15,41 +16,53 @@ import {
   EuiTitle,
   useEuiTheme,
 } from '@elastic/eui';
-import { AttackDiscovery } from '@kbn/elastic-assistant-common';
+import { AttackDiscoveryAlert } from '@kbn/elastic-assistant-common';
 import { css } from '@emotion/react';
+import { useSearchParams } from 'react-router-dom-v5-compat';
 import { useAssistantContext } from '../../assistant_context';
 import { AttackDiscoveryDetails } from './attack_discovery_details';
-import { useFetchAttackDiscovery } from './use_fetch_attack_discovery';
+import { useFindAttackDiscoveries } from './use_find_attack_discoveries';
 import * as i18n from './translations';
 
 interface Props {
-  // TODO use alert id for attack discovery
-  id?: string;
+  id: string;
 }
 
 export const AttackDiscoveryWidget = memo(({ id }: Props) => {
-  const { http, toasts, navigateToApp } = useAssistantContext();
+  const { assistantAvailability, http, navigateToApp } = useAssistantContext();
   const { euiTheme } = useEuiTheme();
-  // TODO fetch by alert id, not connector id. Waiting for Andrew's API updates
-  const connectorId = 'my-gpt4o-ai';
-  const { isFetching, data } = useFetchAttackDiscovery({ connectorId, http, toasts });
-  const [attackDiscovery, setAttackDiscovery] = useState<AttackDiscovery | null>(null);
+
+  const { pathname } = useLocation();
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const { isLoading, data } = useFindAttackDiscoveries({
+    alertIds: [id],
```

Comment:
I'm wondering if this widget should take the `start` and `end` dates as additional props, and pass them to the find API.

I'm also wondering if they should be passed as URL parameters to the Attack discovery page.

Reply:
Yea I was wondering that too... so right now because we don't specify `start` and `end`, the alert summary table could be looking at alerts from 2 months ago, find a discovery from 2 months ago, and then link to attack discovery but the page would be set to the default of past 24 hours, and nothing would show. Is that right?

Reply:
I agree, but I have 2 other issues to sort out today so I may need to loop back to this. Maybe we should merge and follow up with the fix